codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -320,6 +328,24 @@ func (r *reconcilerImpl) Reconcile(ctx {{.contextContext|raw}}, key string) erro
 		}
 	}
 
+	{{if .genDuckLogic}}
+	// Bump observed generation to denote that we have processed this
+	// generation regardless of success or failure.
+	resource.Status.ObservedGeneration = resource.Generation
+
+	if resource.Status.ObservedGeneration != original.Status.ObservedGeneration && reconcileEvent != nil {
+		originalRc := original.Status.GetCondition(apis.ConditionReady)
+		rc := resource.Status.GetCondition(apis.ConditionReady)
+		// if a new generation is observed and reconciliation reported an error event
+		// the reconciler should change the ready state. By default we will set unknown.
+		if equality.Semantic.DeepEqual(originalRc, rc) {
+			logger.Warnw("A reconconiler observed a new generation without updating the resource status")
+			condSet.Manage(&resource.Status).MarkUnknown(
```

Comment:
I don't think you should change the ready condition directly. It needs a dependent condition that cases the unreadiness.